dashboards/component/stat/serializers.py

```diff
@@ -91,25 +94,46 @@ def get_queryset(self, *args, **kwargs) -> QuerySet:
 
         return queryset
 
+    def apply_filters(self, queryset, filters):
+        # Apply filters from FilterComponent
+        if self._meta.filter_component:
+            queryset = self._meta.filter_component.apply_filters(queryset, filters)
+
+        return queryset
+
     @classmethod
     def serialize(cls, **kwargs) -> StatSerializerData:
         raise NotImplementedError
 
+class StatSerializer(BaseStatSerializer):
+
+  def get_queryset(self, *args, **kwargs):
+
+    filter = MyFilter()
+    queryset = super().get_queryset(*args, **kwargs)
+    queryset = filter.filter(queryset, self.context['request'].GET)
+
+    return queryset 
+
 
 class StatSerializer(BaseStatSerializer, asset_definitions.MediaDefiningClass):
     template_name: str = "dashboards/components/stat/stat.html"
 
     class Media:
         js = ("https://unpkg.com/feather-icons", "dashboards/js/icons.js")
-
+    
     def get_value(self) -> Any:
         queryset = self.get_queryset()
         queryset = self.aggregate_queryset(queryset)
         return queryset[self.annotated_field_name]
-
+    
     @classmethod
     def serialize(cls, **kwargs) -> StatSerializerData:
         self = cls()
+        filters = {}  # here we can  obtain filters from the FilterComponent
+        queryset = self.apply_filters(self.get_queryset(), filters)
+        queryset = self.aggregate_queryset(queryset)
+        queryset = self.apply_filters(queryset, filters)
```

Comment:
queryset is not used so neither is the filter.

Reply:
also `filters` is hard coded to an empty dict so how will it ever filter?